グループ作成機能の実装

yosuketokunaga opened the pull request:
#  WHAT
グループ作成機能の実装

#  WHY
グループ管理に必須の機能なため

Commit:
create pull request

Commit:
Create_group＿Editing_screen

・Create_group＿Editing_screen

Commit:
Delete_some_sentences

・Delete_some_sentences

yosuketokunaga commented:
![2017-04-12 19 43 51](https://cloud.githubusercontent.com/assets/26739922/24954098/d81f33b4-1fb8-11e7-923d-4aec4a4712f7.png)

yosuketokunaga commented:
![2017-04-12 19 43 47](https://cloud.githubusercontent.com/assets/26739922/24954100/deaaa31c-1fb8-11e7-946c-f34164c82b82.png)

yosuketokunaga commented:
![2017-04-12 19 48 50](https://cloud.githubusercontent.com/assets/26739922/24954158/2b3c1e90-1fb9-11e7-93cc-e540299de231.png)

yosuketokunaga commented:
![2017-04-12 19 49 03](https://cloud.githubusercontent.com/assets/26739922/24954258/90ae499c-1fb9-11e7-98f4-4b0d7802e11b.png)

Commit:
Correction_a_sentence

・Correct a sentence
render→ redirect_to

ninooo29 commented on app/controllers/groups_controller.rb:
```diff
@@ -1,2 +1,19 @@
+    if @group.save
+      redirect_to :root, notice: 'チャットグループが作成されました。'
+    else
+      render new_group_path, alert: 'チャットグループの作成に失敗しました。'
```
flashメッセージをnowメソッドを使って発生させましょう。
失敗時はrenderを使って記述することが多いのでrenderを使ってみてください。

Commit:
edit_controllers

・edit_controllers

yuya1212h4 commented on app/controllers/groups_controller.rb:
```diff
@@ -8,7 +8,8 @@ def create
     else
-      redirect_to :new_group_path, alert: 'チャットグループの作成に失敗しました。'
+      flash.now[:alert] = 'チャットグループの作成に失敗しました。'
+      render :action => :new
```
=> の記法は使わないように記述しましょう。

yosuketokunaga replied:
renderの記述を修正いたしました。

yuya1212h4 commented on app/views/groups/edit.html.haml:
```diff
@@ -0,0 +1,25 @@
+  %form#edit_chat_group_22.edit_chat_group{"accept-charset" => "UTF-8", :action => "/chat_groups/22", :method => "post"}
+    %input{:name => "utf8", :type => "hidden", :value => "✓"}/
+    %input{:name => "_method", :type => "hidden", :value => "patch"}/
+    %input{:name => "authenticity_token", :type => "hidden", :value => "U5ABp0/UobRQTSwsR0V40bX1w3t0rZ0CTqrVyLecD+L6sdhBRO/gVcKy7w/Rgxnc7Hf0Ts39996dal3to5fTag=="}/
```
formタグやinputタグは使わずに、ヘルパーを使用して頂けるようにお願いします。
ハッシュロケットも使用しないようにお願いします。

yosuketokunaga replied:
このpullrequestでは作成機能の実装で出しているので、
edit.html.hamlについては別のpullrequestで対応させていただきます。
ご確認お願いいたします。

yuya1212h4 replied:
かしこまりました。
その際に修正の方はお願い致します。

Commit:
・Correct a sentence

・Correct a sentence

yosuketokunaga commented:
renderの記述を修正いたしました。

このpullrequestでは作成機能の実装で出しているので、
edit.html.hamlについては別のpullrequestで対応させていただきます。
ご確認お願いいたします。

yuya1212h4 commented on config/routes.rb:
```diff
@@ -2,4 +2,5 @@
   devise_for :users
   root  'messages#index'
   resources :messages, only: [:index]
+  resources :groups, only: [:new, :create, :edit, :update]
```
現段階で必要なものだけを指定して頂けたらと思います。

yosuketokunaga replied:
修正いたしました。

Commit:
・correct a sentence

・correct a sentence

yusukeyamane commented on app/views/groups/new.html.haml:
```diff
@@ -0,0 +1,29 @@
+      .chat-group-form__field--right
+        %ul#group_user_list
+          %li.box
+            （ログイン中のユーザー）
```
こちらの記述は必要でしょうか？

yosuketokunaga replied:
不要ですね。削除いたしました。

yusukeyamane commented on app/controllers/groups_controller.rb:
```diff
@@ -1,2 +1,20 @@
+    end
+  end
+
+  private
```
コーディング規約に従いprivateの前後は一行ずつ改行しましょう

yosuketokunaga replied:
private前後に改行を入れました。

yosuketokunaga commented:
![2017-04-13 16 55 39](https://cloud.githubusercontent.com/assets/26739922/24995547/29bf1be2-206a-11e7-87df-becf8af3b316.png)

Commit:
・Correct a sentences

・add new_line before and after “private”
・delete sentences about who log in now from new.html.haml
